src/RecurringAllowanceManager.sol

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IERC1271} from "openzeppelin-contracts/contracts/interfaces/IERC1271.sol";
+import {IERC20} from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {CoinbaseSmartWallet} from "smart-wallet/CoinbaseSmartWallet.sol";
+
+/// @title RecurringAllowanceManager
+///
+/// @notice Allow spending native and ERC20 tokens with a recurring allowance.
+///
+/// @dev Allowance and spend values capped at uint160 ~ 1e48.
+///
+/// @author Coinbase (https://github.com/coinbase/smart-wallet-permissions)
+contract RecurringAllowanceManager {
+    /// @notice A recurring allowance for an external spender to withdraw an account's tokens.
+    struct RecurringAllowance {
+        /// @dev Smart account this recurring allowance is valid for.
+        address account;
+        /// @dev Entity that can spend user funds.
+        address spender;
```

Comment:
nit: should this be like `requester` or `authority` or something? This will be required as msg.sender right?

Reply:
ahh idk maybe spender is good

Reply:
I think spender is the clearest because the primary action is spending!